content/en/user-guide/aws/elastic-compute-cloud/index.md

```diff
@@ -151,11 +156,18 @@ Similar to the setup in production AWS, the user data content is stored at `/var
 Any execution of this data is recorded in the `/var/log/cloud-init-output.log` file.
 {{< /alert >}}
 
-You can also setup an SSH connection to the locally emulated EC2 instance using the IP address.
-Run the following command to setup an SSH connection:
+### Connecting via SSH
 
+You can also set up an SSH connection to the locally emulated EC2 instance using the instance IP address.
+
+This section assumes that you have created or imported an SSH keypair named `my-key` (see [instructions above](#create-a-key-pair)). When running the EC2 instance, make sure to pass the `--key-name` parameter to the command:
 {{< command >}}
-$ ssh -p 22 -i test.pem 172.17.0.4
+$ awslocal ec2 run-instances --key-name my-key ...
+{{< /command >}}
+
+Once the instance is up and running, we can use the `ssh` command to set up an SSH connection. Assuming the instance is available under `127.0.0.1:12862` (as per the LocalStack log output), use this command:
+{{< command >}}
+$ ssh -p 12862 -i ~/.ssh/id_rsa root@127.0.0.1
 {{< /command >}}
 
```

Comment:
Sneaking in a hint for an error message that is a bit misleading.

```suggestion

{{< alert title="Hint" color="info">}}
If `ssh` command throws an error like so:

> Warning: Identity file key.pem not accessible: No such file or directory.

Please make sure that the key file has a restrictive `0400` permission as illustrated [here](#create-a-key-pair).
{{< /alert >}}

```

Reply:
Thanks for adding this one @viren-nadkarni ! Seems like this error message indicates that the file doesn't exist at all, right?

On MacOS, I'm seeing something like this for too permissive key files:
```
Permissions 0755 for '/Users/whummer/.ssh/id_rsa' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
Load key "/Users/whummer/.ssh/id_rsa": bad permissions
```

Reply:
Slightly updated your suggestion, to cover both cases.. 👍